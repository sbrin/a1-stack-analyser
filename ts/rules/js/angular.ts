import { register } from '../../register.js';

register({
  tech: 'angular',
  name: 'Angular',
  type: 'language',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
