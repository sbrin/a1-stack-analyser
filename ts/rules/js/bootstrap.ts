import { register } from '../../register.js';

register({
  tech: 'bootstrap',
  name: 'Bootstrap',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'bootstrap' }],
});
