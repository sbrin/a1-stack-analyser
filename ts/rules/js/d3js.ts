import { register } from '../../register.js';

register({
  tech: 'd3js',
  name: 'D3.js',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'd3' },
    { type: 'npm', name: 'd3-shape' },
    { type: 'npm', name: '@types/d3' },
    { type: 'npm', name: '@types/d3-shape' },
  ],
});
