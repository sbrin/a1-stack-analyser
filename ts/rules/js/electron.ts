import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'electron' },
    { type: 'githubAction', name: 'samuelmeuli/action-electron-builder' },
  ],
});
