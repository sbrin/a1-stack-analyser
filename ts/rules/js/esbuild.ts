import { register } from '../../register.js';

register({
  tech: 'esbuild',
  name: 'Esbuild',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'esbuild' }],
});
