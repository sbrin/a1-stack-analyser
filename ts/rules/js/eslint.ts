import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: ['.eslintrc', '.eslintrc.cjs', '.eslintrc.json', '.eslintrc.js'],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
