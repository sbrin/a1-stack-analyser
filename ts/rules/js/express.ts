import { register } from '../../register.js';

register({
  tech: 'express',
  name: 'Express',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'express' }],
});
