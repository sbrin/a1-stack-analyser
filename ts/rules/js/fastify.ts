import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'fastify' },
    { type: 'npm', name: '@opentelemetry/instrumentation-fastify' },
  ],
});
