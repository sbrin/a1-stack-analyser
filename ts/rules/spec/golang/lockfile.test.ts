import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';

import '../../../autoload.js';

describe('golang (lockfile)', () => {
  it('should match everything', async () => {
    const deps = [];
    for (const item of rawList) {
      if (item.type !== 'dependency' || item.ref.type !== 'golang') {
        continue;
      }

      const example = 'example' in item.ref ? item.ref.example : item.ref.name;
      deps.push(`	${example} v1.0.0`);
    }
    const lockfile: string[] = [
      `module github.com/specfy/specfy

go 1.19
require (
${deps.join('\n')}
)`,
    ];

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['go.mod'],
        },
        files: {
          '/go.mod': lockfile.join(''),
        },
      }),
    });

    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toMatchSnapshot();
    expect(Array.from(merged.dependencies).sort()).toMatchSnapshot();
  });
});
