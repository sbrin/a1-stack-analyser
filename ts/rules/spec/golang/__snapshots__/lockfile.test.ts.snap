// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`golang (lockfile) > should match everything 1`] = `
[
  "aws",
  "aws.apigateway",
  "aws.athena",
  "aws.cloudformation",
  "aws.cloudfront",
  "aws.cloudsearch",
  "aws.cloudwatch",
  "aws.codebuild",
  "aws.documentdb",
  "aws.dynamodb",
  "aws.ebs",
  "aws.ec2",
  "aws.ecr",
  "aws.ecs",
  "aws.efs",
  "aws.eks",
  "aws.elasticache",
  "aws.glacier",
  "aws.glue",
  "aws.kinesis",
  "aws.kms",
  "aws.lambda",
  "aws.lightsail",
  "aws.memorydb",
  "aws.mq",
  "aws.neptune",
  "aws.opensearch",
  "aws.polly",
  "aws.rds",
  "aws.redshift",
  "aws.rekognition",
  "aws.s3",
  "aws.sagemaker",
  "aws.secretsmanager",
  "aws.ses",
  "aws.sfn",
  "aws.sns",
  "aws.sqs",
  "aws.timestream",
  "aws.translate",
  "azure",
  "azure.cosmosdb",
  "azure.mariadb",
  "azure.mysql",
  "azure.postgres",
  "azure.redis",
  "azure.sql",
  "brevo",
  "clickhouse",
  "elasticsearch",
  "gcp",
  "gcp.aiplatform",
  "gcp.appengine",
  "gcp.artifactregistry",
  "gcp.bigquery",
  "gcp.bigtable",
  "gcp.cloudbuild",
  "gcp.cloudrun",
  "gcp.dataflow",
  "gcp.dataproc",
  "gcp.datastore",
  "gcp.dialogflow",
  "gcp.functions",
  "gcp.gce",
  "gcp.gcs",
  "gcp.kms",
  "gcp.language",
  "gcp.logging",
  "gcp.maps",
  "gcp.memorystore",
  "gcp.pubsub",
  "gcp.secretmanager",
  "gcp.spanner",
  "gcp.speech",
  "gcp.tasks",
  "gcp.translate",
  "gcp.vision",
  "gitguardian",
  "golang",
  "hyperdx",
  "lago",
  "launchdarkly",
  "lemonsqueezy",
  "logstash",
  "mailgun",
  "meilisearch",
  "milvusdb",
  "nats",
  "novu",
  "optimizely",
  "papertrail",
  "paypal",
  "pirschanalytics",
  "posthog",
  "prometheus",
  "qdrant",
  "qovery",
  "questdb",
  "rethinkdb",
  "rollbar",
  "sendgrid",
  "slack",
  "splitio",
  "stripe",
  "tdengine",
  "telegram",
  "twilio",
  "typesense",
  "zipkin",
]
`;

exports[`golang (lockfile) > should match everything 2`] = `
[
  [
    "golang",
    "cloud.google.com/go/aiplatform",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/appengine",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/artifactregistry",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/bigquery",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/bigtable",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/cloudbuild",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/compute",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/dataflow",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/dataproc",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/datastore",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/dialogflow",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/functions",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/kms",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/language",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/logging",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/maps",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/pubsub",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/redis",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/run",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/secretmanager",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/spanner",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/speech",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/storage",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/tasks",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/translate",
    "v1.0.0",
  ],
  [
    "golang",
    "cloud.google.com/go/vision",
    "v1.0.0",
  ],
  [
    "golang",
    "contrib.go.opencensus.io/exporter/zipkin",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cosmos/armcosmos",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/mariadb/armmariadb",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/mysql/armmysql",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/postgresql/armpostgresql",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/redis/armredis/v2",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/sql/armsql",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/ClickHouse/clickhouse-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/Gaardsholt/go-gitguardian",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/NdoleStudio/lemonsqueezy-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/apache/thrift",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/apigateway",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/athena",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/cloudformation",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/cloudfront",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/cloudsearch",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/cloudwatch",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/codebuild",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/docdb",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/dynamodb",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/ebs",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/ec2",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/ecr",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/ecs",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/efs",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/eks",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/elasticache",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/firehose",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/glacier",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/glue",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/kinesis",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/kms",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/lambda",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/lightsail",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/memorydb",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/mq",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/neptune",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/opensearch",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/polly",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/rds",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/redshift",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/rekognition",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/s3",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/sagemaker",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/secretsmanager",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/ses",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/sfn",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/sns",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/sqs",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/timestream",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/aws/aws-sdk-go-v2/service/translate",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/bshuster-repo/logrus-logstash-hook",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/elastic/beats/libbeat/outputs/logstash",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/elastic/go-elasticsearch",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/getbrevo/brevo-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/getlago/lago-go-client",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/go-acme/lego/v4/cmd/lego",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/go-kit/kit/metrics/prometheus",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/go-kit/kit/tracing/zipkin",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/go-telegram-bot-api/telegram-bot-api/v5",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/golangci/golangci-lint",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/grpc-ecosystem/go-grpc-prometheus",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/hashicorp/terraform-plugin-sdk/v2",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/hyperdxio/hyperdx-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/launchdarkly/go-sdk-common/v3",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/mailgun/mailgun-go/v4",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/mailhog/MailHog",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/markbates/goth/providers/paypal",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/meilisearch/meilisearch-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/milvus-io/milvus-sdk-go/v2",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/nats-io/nats.go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/nats-io/stan.go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/novuhq/go-novu",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/openzipkin-contrib/zipkin-go-opentracing",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/openzipkin/zipkin-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/optimizely/go-sdk",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/papertrail/go-tail/follower",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/pirsch-analytics/pirsch-go-sdk/v2",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/posthog/posthog-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/prometheus/client_golang/prometheus",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/qdrant/go-client",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/qovery/qovery-client-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/questdb/go-questdb-client",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/rethinkdb/rethinkdb-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/rollbar/rollbar-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/sendgrid/sendgrid-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/slack-go/slack",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/splitio/go-client/v6",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/stripe/stripe-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/taosdata/driver-go/v3/taosSql",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/tebeka/selenium",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/twilio/twilio-go",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/typesense/typesense-go/typesense",
    "v1.0.0",
  ],
  [
    "golang",
    "github.com/wamuir/graft/tensorflow",
    "v1.0.0",
  ],
  [
    "golang",
    "go.k6.io/xk6/cmd/xk6ver",
    "v1.0.0",
  ],
  [
    "golang",
    "go.opentelemetry.io/otel/exporters/zipkin",
    "v1.0.0",
  ],
  [
    "golang",
    "goveralls",
    "v1.0.0",
  ],
]
`;
